### PR TITLE
201909/demo forward ported onto develop

### PR DESCRIPTION
**Pull request by pjkundert**
Added the holofuel-demo-configure script, and its dependencies to the develop branch
o Convert to use holo-init (renamed from holoportos-initialize)
o Clean up the build of hclient and holofuel-app

### CODE REVIEW

File: overlays/holo-nixpkgs/holofuel-demo-configure/holofuel-demo-configure.sh [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+# 0) Create a Deterministic Holo Configuration
+# 
+# The Holo Admin password will be chosen at runtime, and displayed to the person running this
+# script.  It will be unique to this demo instance.
+#
+# TODO: generate password from /etc/machineid, so it is deterministic per-install
+#
+
+password-from-machine-id() {
+    # Take the hex /etc/machine-id, convert to base-6 dice rolls on single lines, generate a
```
Comment (yegortimoshenko) [inline overlays/holo-nixpkgs/holofuel-demo-configure/holofuel-demo-configure.sh]: Double indentation here.

File: overlays/holo-nixpkgs/holofuel-demo-configure/holofuel-demo-configure.sh [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+HOLOFUEL_DNA_HASH=QmcqAKFLP6WrjWghWVzrgnoa72EWu211C7Fu2F1FwRMU1k
+
+HOLOFUEL_DEMO_SITE=holofuel-demo.holohost.net
+
+HOLOCHAIN_DNA_DIR=/var/lib/holo-envoy/.holochain/holo/dnas
+
+set -euo pipefail
+shopt -s extglob nullglob
+
+PATH=@path@:$PATH
```
Comment (yegortimoshenko) [inline overlays/holo-nixpkgs/holofuel-demo-configure/holofuel-demo-configure.sh]: This doesn't seem to be used.

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,3 @@
 *.qcow2
 /result*
+*~
```
[the remark this answers is not in the document]
Comment (pjkundert) [reply .gitignore]: Great idea!

File: overlays/holo-nixpkgs/hclient/default.nix [hunk 4]
```diff
@@ -10,7 +11,7 @@ stdenv.mkDerivation rec {
     sha256 = "1w1sljpblf9bj66wagscn7drjdzz9lby7d0y39ai91xq9mlg67xy";
   };
 
-  nativeBuildInputs = [ nodejs ];
+  nativeBuildInputs = [ nodejs utillinux ];
```
[the remark this answers is not in the document]
Comment (pjkundert) [reply overlays/holo-nixpkgs/hclient/default.nix]: It doesn't find it, but at least lscpu is there...

File: overlays/holo-nixpkgs/default.nix [hunk 5]
```diff
@@ -112,33 +112,50 @@ in
     servicelogger = callPackage ./dna-packages/servicelogger {};
   };
 
+  # 
+  # HoloPort Runtime and Hardware Support
```
[the remark this answers is not in the document]
Comment (pjkundert) [reply overlays/holo-nixpkgs/default.nix]: Perhaps you're right; lets clean this up as we continue modifying holo-nixpkgs/default.nix

**Comment by pjkundert** [comment]
Integrated holo-config into holofuel-demo-configure

**Comment by yegortimoshenko** [comment]
@pjkundert I didn't notice this was merged, we require at least one approving review (from anyone, but me and @samrose are best placed to do that in Holo Nixpkgs as of today).

File: aarch64-qemu.sh [hunk 6]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+# 
+# Run a Debian Installer iso, then boot
```
Comment (yegortimoshenko) [inline aarch64-qemu.sh]: This doesn't belong in Holo Nixpkgs. We can port `holoportos-shell` to boot an aarch64-linux QEMU box.

File: overlays/holo-nixpkgs/default.nix [hunk 7]
```diff
@@ -112,33 +112,49 @@ in
...
   holoport-hardware-test = callPackage ./holoport-hardware-test {};
 
   holoport-nano-dtb = callPackage ./holoport-nano-dtb {
     linux = linux_latest;
   };
 
-  holo-init = callPackage ./holo-init {
+  # 
+  # Holo Packages
+  #
```
Comment (yegortimoshenko) [inline overlays/holo-nixpkgs/default.nix]:
```diff
-  # 
-  # Holo Packages
-  # 
```

**Pull request merged**